parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/internal/ConsumerManager.java

```diff
@@ -94,7 +94,21 @@ public void commitSync(final Map<TopicPartition, OffsetAndMetadata> offsetsToSen
         noWakeups++;
         while (inProgress) {
             try {
-                consumer.commitSync(offsetsToSend);
+                while(true) {
+                    try {
+                        consumer.commitSync(offsetsToSend);
+                        // break when offset commit is okay. Do not throw exception to main threads
+                        break;
+                    } catch(Throwable t) {
```

Comment:
Wouldn't it be better to explicitly specify the exceptions that are retriable? You wouldn't want to be stuck in a commit loop if the call has no chance of ever succeeding.